pybricks/media/pb_type_image.c

```diff
@@ -0,0 +1,353 @@
+    self->buffer = m_malloc0(width * height * sizeof(uint8_t));
+    self->is_display = false;
+    pbio_image_init(&self->image, self->buffer, width, height, width);
+
+    return MP_OBJ_FROM_PTR(self);
+}
+static MP_DEFINE_CONST_FUN_OBJ_KW(pb_type_Image_empty_fun_obj, 0, pb_type_Image_empty);
+static MP_DEFINE_CONST_STATICMETHOD_OBJ(pb_type_Image_empty_obj, MP_ROM_PTR(&pb_type_Image_empty_fun_obj));
+
+static void pb_type_Image_attr(mp_obj_t self_in, qstr attr, mp_obj_t *dest) {
+    // Read only
+    if (dest[0] == MP_OBJ_NULL) {
+        pb_type_Image_obj_t *self = MP_OBJ_TO_PTR(self_in);
+        if (attr == MP_QSTR_width) {
+            dest[0] = mp_obj_new_int(self->image.width);
+            return;
+        }
+        if (attr == MP_QSTR_height) {
+            dest[0] = mp_obj_new_int(self->image.height);
+            return;
+        }
+    }
+    // Attribute not found, continue lookup in locals dict.
+    dest[1] = MP_OBJ_SENTINEL;
+}
+
+static mp_obj_t pb_type_Image_clear(mp_obj_t self_in) {
+    pb_type_Image_obj_t *self = MP_OBJ_TO_PTR(self_in);
+
+    pbio_image_fill(&self->image, 0);
+
+    if (self->is_display) {
+        pbdrv_display_update();
+    }
+
+    return mp_const_none;
+}
+MP_DEFINE_CONST_FUN_OBJ_1(pb_type_Image_clear_obj, pb_type_Image_clear);
+
+static mp_obj_t pb_type_Image_load_image(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    PB_PARSE_ARGS_METHOD(n_args, pos_args, kw_args,
+        pb_type_Image_obj_t, self,
+        PB_ARG_REQUIRED(source));
+
+    pb_assert_type(source_in, &pb_type_Image);
+    pb_type_Image_obj_t *source = MP_OBJ_TO_PTR(source_in);
+
+    int x = (self->image.width - source->image.width) / 2;
+    int y = (self->image.height - source->image.height) / 2;
+
+    pbio_image_fill(&self->image, 0);
+    pbio_image_draw_image(&self->image, &source->image, x, y);
+
+    if (self->is_display) {
+        pbdrv_display_update();
+    }
+
+    return mp_const_none;
+}
+static MP_DEFINE_CONST_FUN_OBJ_KW(pb_type_Image_load_image_obj, 1, pb_type_Image_load_image);
+
+static mp_obj_t pb_type_Image_draw_image(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    PB_PARSE_ARGS_METHOD(n_args, pos_args, kw_args,
+        pb_type_Image_obj_t, self,
+        PB_ARG_REQUIRED(x),
+        PB_ARG_REQUIRED(y),
+        PB_ARG_REQUIRED(source),
+        PB_ARG_DEFAULT_NONE(transparent));
+
+    mp_int_t x = mp_obj_get_int(x_in);
+    mp_int_t y = mp_obj_get_int(y_in);
+    pb_assert_type(source_in, &pb_type_Image);
+    pb_type_Image_obj_t *source = MP_OBJ_TO_PTR(source_in);
+    bool transparent = false;
+    int transparent_value;
+    if (transparent_in != mp_const_none) {
+        transparent = true;
+        transparent_value = get_color(transparent_in);
+    }
+
+    if (!transparent) {
+        pbio_image_draw_image(&self->image, &source->image, x, y);
+    } else {
+        pbio_image_draw_image_transparent(&self->image, &source->image, x, y, transparent_value);
+    }
+
```

Comment:
```suggestion
    if (transparent_in != mp_const_none) {
        int transparent_value = get_color(transparent_in);
        
        pbio_image_draw_image_transparent(&self->image, &source->image, x, y, transparent_value);
    } else {
        pbio_image_draw_image(&self->image, &source->image, x, y);
    }

```

Or just combine the if statements to avoid it altogether.